### PR TITLE
fix: add chainId to txn params

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What was wrong?

I noticed even when caching `chain_id`, the request occurs multiple times. I figured out this happens because I am calling `estimate_gas()` and passing it a transaction and the RPC method's defined txn schema does not inclue `chainId` so it makes the request againt to add the chainId at some point (this is only deduced).. By adding `chainId` to the params, I can avoid an extra RPC call to get the chain ID`

### How was it fixed?

Add `'chainId': 'uint',` in `TRANSACTION_PARAMS_ABIS`.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture
baby_bats

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: web3/_utils/rpc_abi.py [hunk 1]
```diff
@@ -163,6 +163,7 @@ class RPC:
     'nonce': 'uint',
     'to': 'address',
     'value': 'uint',
+    'chainId': 'uint',
```
Comment (unparalleled-js) [inline web3/_utils/rpc_abi.py:166]: Is there a way to make this optional?

**Comment by kclowes** [comment]
Nice catch! I don't think there is a way to indicate that it's optional, but I don't think we do any checks on it. Lmk if you want help debugging the tests. It looks like they're failing because a hex-to-int conversion isn't happening for the error message in the validation middleware.

**Comment by unparalleled-js** [comment]
> It looks like they're failing because a hex-to-int conversion isn't happening for the error message in the validation middleware.

I believe I got it!

**Review by kclowes — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**